### ISSUE TITLE
[gazebo_plugins] Build fail: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’

### ISSUE THREAD

**Issue by fmessmer** [ros-simulation/gazebo_ros_pkgs#321]
With today's updates on `indigo-devel`, I get the following build error:
(I'm using regular deb for gazebo2 `2.2.3-1~trusty`)

```
Scanning dependencies of target gazebo_ros_skid_steer_drive
[  9%] Building CXX object gazebo_ros_pkgs/gazebo_plugins/CMakeFiles/gazebo_ros_skid_steer_drive.dir/src/gazebo_ros_skid_steer_drive.cpp.o
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp: In member function ‘virtual void gazebo::GazeboRosSkidSteerDrive::Load(gazebo::physics::ModelPtr, sdf::ElementPtr)’:
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:259:25: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[LEFT_FRONT]->SetParam("fmax", 0, torque);
                         ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:260:26: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[RIGHT_FRONT]->SetParam("fmax", 0, torque);
                          ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:261:24: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[LEFT_REAR]->SetParam("fmax", 0, torque);
                        ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:262:25: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[RIGHT_REAR]->SetParam("fmax", 0, torque);
                         ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp: In member function ‘virtual void gazebo::GazeboRosSkidSteerDrive::UpdateChild()’:
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:311:27: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[LEFT_FRONT]->SetParam("vel", 0, wheel_speed_[LEFT_FRONT] / (wheel_diameter_ / 2.0));
                           ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:312:28: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[RIGHT_FRONT]->SetParam("vel", 0, wheel_speed_[RIGHT_FRONT] / (wheel_diameter_ / 2.0));
                            ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:313:26: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[LEFT_REAR]->SetParam("vel", 0, wheel_speed_[LEFT_REAR] / (wheel_diameter_ / 2.0));
                          ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:314:27: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[RIGHT_REAR]->SetParam("vel", 0, wheel_speed_[RIGHT_REAR] / (wheel_diameter_ / 2.0));
                           ^
make[2]: *** [gazebo_ros_pkgs/gazebo_plugins/CMakeFiles/gazebo_ros_skid_steer_drive.dir/src/gazebo_ros_skid_steer_drive.cpp.o] Error 1
make[1]: *** [gazebo_ros_pkgs/gazebo_plugins/CMakeFiles/gazebo_ros_skid_steer_drive.dir/all] Error 2
make: *** [all] Error 2
make: INTERNAL: Exiting with 5 jobserver tokens available; should be 4!
Invoking "make -j4 -l4" failed
```

**Comment by scpeters** [comment]
My fault! I merged #315, which requires gazebo 3+; I'll add ifdef's for gazebo2

**Comment by scpeters** [comment]
Should be fixed by #322, thanks for the report! Sorry for breaking it.
